go/tasks/pluginmachinery/ioutils/in_memory_output_reader.go

```diff
@@ -47,10 +48,15 @@ func (r InMemoryOutputReader) DeckExists(_ context.Context) (bool, error) {
 	return r.DeckPath != nil, nil
 }
 
-func NewInMemoryOutputReader(literals *core.LiteralMap, DeckPath *storage.DataReference, err *io.ExecutionError) InMemoryOutputReader {
+func (r InMemoryOutputReader) SpanExists(_ context.Context) (bool, error) {
+	return r.SpanPath != nil, nil
+}
+
+func NewInMemoryOutputReader(literals *core.LiteralMap, DeckPath *storage.DataReference, SpanPath *storage.DataReference, err *io.ExecutionError) InMemoryOutputReader {
```

Comment:
Could we add a new class called `InMemoryOutputReaderWithSpan`?  Spotify is using this class internally, so we can't change it. https://github.com/flyteorg/flyteplugins/pull/268#issuecomment-1204913608